source/equations.py

```diff
@@ -128,17 +138,122 @@ def phasespace_eval(self, t, r) -> tuple:
         """
         return tuple(eqn.eval_rhs(t=t, r=r) for eqn in self.equations)
 
+    def eval_jacobian(self, r):
+        """
+        Evaluates the symbolic Jacobian of the system at the point r
+        """
+        # The subs method substitutes one symbol or value for another
+        jacobian = self.jacobian.subs(self.params)
+        jacobian = jacobian.subs(list(zip(self.system_coord_symbols, r)))
+        return jacobian
+
+    def show_jacobian(self, eval=False, r=None):
+        """
+        Prints Jacobian. May be evaluated at a point first, or printed symbolically
+        """
+
+        jacobian = self.eval_jacobian(r) if eval else self.jacobian
+        sp.pprint(jacobian)
+
+    def eigenvects(self, r=None):
+        """
+        Calculates the eigenvalues and eigenvectors of the system's Jacobian.
+        Return list of triples (eigenval, multiplicity, eigenspace).
+        """
+
+        jacobian = self.eval_jacobian(r) if r is not None else self.jacobian
+        return jacobian.eigenvects(simplify=True)
+
+    def calc_fixed_points(self):
+        """
+        Returns a set of fixed points as tuples.
+        """
+
+        eqns = [eqn.expr for eqn in self.equations]
+        eqns = [eqn.subs(self.params) for eqn in eqns]
+        _, fps = sp.solve(eqns, self.system_coord_symbols, set=True)
+
+        fps_as_cmplx = {tuple(complex(z) for z in fp) for fp in fps}
+        fps_rounded = {tuple(round_complex(z, 3) for z in fp) for fp in fps_as_cmplx}
+        fps_no_cmplx = {
+            fp for fp in fps_rounded if all(not abs(z.imag) > 0 for z in fp)
+        }
+        fps_real = {tuple(z.real for z in fp) for fp in fps_no_cmplx}
```

Comment:
Does this need four lines? Also, curious, why use sets?

Reply:
If you mean the formatting, then that was the formatter's work.

I used sets because the output of the Sympy solve function that I found most useful returns a set, so I figured that I'd keep the code consistent with this output. Also, it makes sense to keep a collection of FPs as a set as ordering and duplication of FPs is irrelevant in such a collection.